blog/docs/articles/demystifying-activator-on-path.md

```diff
@@ -0,0 +1,182 @@
+When in proxy mode Activator is on the data path and it will stay on path until certain conditions are met.
+When these conditions are met Activator is removed from the path and the service transitions to serve mode.
+Although it was not always like that when a service scales from/to zero activator is added by default to the path.
+This default setting often confuses users for reasons we will see next as it is possible that activator will not
+be removed unless enough capacity is available. This is intended as one of the Activator's roles is to offer backpressure capabilities so that it can make sure that a service could serve the traffic.
+
+
+## Background
+
+The default pod autoscaler in Knative (KPA) is a sophisticated algorithm that uses metrics from pods to
+calculate metrics and make scaling decisions. Let's see in detail what happens when a new Knative service is created.
+
+Once the user creates a new service the corresponding Knative reconciler creates a Knative Configuration and a Knative Route for that service. Then the Configuration reconciler creates a Revision resource and
+the reconciler for the latter will create a Pod Autoscaler(PA) resource along with the K8s deployment for the service.
+The Route reconciler will create the ingress resource that will be picked up by the Knative net-* components responsible
+for managing traffic locally in the cluster and externally to the cluster.
+
+Now the creation of the PA earlier triggers the KPA reconciler which goes through certain steps in order to setup an autoscaling configuration for the revision:
+
+- creates an internal Decider resource that holds the initial desired scale in `decider.Status.DesiredScale`and
+sets up a pod scaler via the multi-scaler component. The pod scaler every two seconds calculates a new Scale
+result and makes a decision based on the condition `decider.Status.DesiredScale != sRes.DesiredPodCount` whether to trigger a new reconciliation for the KPA reconciler. Goal is the KPA to get the latest scale result.
+
+- creates a Metric resource that triggers the metrics collector controller to setup a scraper for the revision pods.
+
+- calls a scale method that decides the number of wanted pods and also updates the revision deployment
+
+- creates/updates a ServerlessService (SKS) that holds info about the operation mode (proxy or serve) and stores the activators used in proxy mode. That SKS create/update event triggers a reconciliation for the SKS from its specific controller that creates the required public and private K8s services so traffic can be routed to the K8s deployment.
+This in combination with the networking setup done by the net-* components is the
+
+- updates the PA and reports the active and wanted pods in its status
+
+## Capacity and Operation Modes in Practice
+
+As described earlier Activator will be removed if enough capacity is available and there is an invariant that needs to
```

Comment:
Can you reword this a bit? its pretty hard to read and comprehend

Reply:
yeah there are some diagrams missing, visually it is easier to understand but in general people need to go through some concepts for good or bad so it can be simplified to some extent.